oneview_redfish_toolkit/blueprints/computer_system.py

```diff
@@ -200,3 +203,100 @@ def _get_oneview_resource(uuid):
+    if not request.is_json:
+        abort(status.HTTP_400_BAD_REQUEST,
+              "The request content should be a valida JSON")
+
+    body = request.get_json()
+
+    #  TODO(@ricardogpsf) use redfish validation instead of this intern class
+    class ComposedSystem(RedfishJsonValidator):
+        SCHEMA_NAME = 'ComputerSystem'
+
+        def __init__(self, composed_system):
+            super().__init__(self.SCHEMA_NAME)
+
+            self.redfish = composed_system
+
+        def validate(self):
+            self._validate()
+
+    try:
+        resource = ComposedSystem(body)
+        resource.validate()
+
+        blocks = resource.redfish["Links"]["ResourceBlocks"]
+        block_ids = \
+            list(map(lambda b: b["@odata.id"].split("/")[-1], blocks))
+
+        # Should contain only one computer system entry
+        system_blocks = _get_system_resource_blocks(block_ids)
+        if not system_blocks:
+            raise ValidationError(
+                "Should there is a Compute Resource Block")
+
+        # Check network block id with the Id attribute in the request
+        network_blocks = _get_network_resource_blocks(block_ids)
+        if not (network_blocks and body["Id"] in network_blocks[0]["uri"]):
+            raise ValidationError(
+                "Should there is a valid Network Resource Block")
+
+        # It can contain zero or more Storage Block
+        storage_blocks = _get_storage_resource_blocks(block_ids)
+
+        spt = g.oneview_client.server_profile_templates.get(body["Id"])
+
+        server_profile = ComputerSystem.build_server_profile(
+            body["Name"], spt, system_blocks, network_blocks, storage_blocks)
+
+        result = g.oneview_client.server_profiles.create(server_profile)
+
+    except ValidationError as e:
+        abort(status.HTTP_400_BAD_REQUEST, e.message)
+    except KeyError as e:
+        abort(status.HTTP_400_BAD_REQUEST,
+              "Trying access an invalid key {}".format(e.args))
+
+    location_uri = ComputerSystem.BASE_URI + "/" + result["uuid"]
+
+    return Response(status=status.HTTP_201_CREATED,
+                    headers={"Location": location_uri},
+                    mimetype="application/json")
+
+
+def _get_system_resource_blocks(ids):
+    return _get_resource_block_data(
+        g.oneview_client.server_hardware.get, ids)
+
+
+def _get_network_resource_blocks(ids):
+    return _get_resource_block_data(
+        g.oneview_client.server_profile_templates.get, ids)
+
+
+def _get_storage_resource_blocks(ids):
+    drive_ids = ["/rest/drives/" + i for i in ids]
+
+    return _get_resource_block_data(
+        g.oneview_client.index_resources.get, drive_ids)
+
+
+def _get_resource_block_data(func, uuids):
+    resources = list()
+
+    for uuid in uuids:
+        try:
+            resource = func(uuid)
+
+            resources.append(resource)
+        except HPOneViewException as e:
+            if e.oneview_response["errorCode"] == 'RESOURCE_NOT_FOUND':
```

Comment:
You are not using this method for server profiles then that's fine. But if it receives a "server_profiles.get" as "func" parameter, it must check for "ProfileNotFoundException" for errorCode as well. Maybe just a comment remembering it would be fine.

Reply:
😨 This is bad, but thruth. The message come with different "errorCode". 😞 
I'll comment this. ;)